### ISSUE TITLE
[Bug]: pg_dump fails due to permission issue in 2.10.0

### ISSUE THREAD

**Issue by lkshminarayanan** [timescale/timescaledb#5449]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Backup

### What happened?

PR #5218 restricts the permissions to the `_timescaledb_internal.job_errors` table so that the users can only view the job
errors log through the view `timescaledb_information.job_errors`. The `_timescaledb_internal.job_errors` is also configured to be included in a pg_dump. But due to the above said permissions updates, pg_dump is unable to lock the table during backup. It fails with the following error :

```
pg_dump: error: query failed: ERROR:  permission denied for table job_errors
pg_dump: detail: Query was: LOCK TABLE _timescaledb_internal.job_errors IN ACCESS SHARE MODE
```

If the `timescaledb_information.job_errors` is to be restricted for the users, it should be configured to be excluded from pg_dump as well.

### TimescaleDB version affected

2.10.0

### PostgreSQL version used

14.5

### What operating system did you use?

Ubuntu 22.04

### What installation method did you use?

Source

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
$ pg_dump -U testuser -v testdb
pg_dump: last built-in OID is 16383
pg_dump: reading extensions
pg_dump: identifying extension members
pg_dump: reading schemas
pg_dump: reading user-defined tables
pg_dump: error: query failed: ERROR:  permission denied for table job_errors
pg_dump: detail: Query was: LOCK TABLE _timescaledb_internal.job_errors IN ACCESS SHARE MODE
```

### How can we reproduce the bug?
Setup the user and database
```sql
-- Create a database and a new user
CREATE DATABASE testdb;
CREATE USER testuser;
GRANT ALL ON DATABASE testdb TO testuser;
\c testdb
GRANT ALL ON SCHEMA public TO testuser;
-- Connect to testdb as testuser
\c testdb testuser
-- Create the timescale extension and table
CREATE EXTENSION timescaledb;
CREATE TABLE t1 (a int);
```
Try taking a backup of testdb as testuser and observe the failure
```bash
pg_dump -U testuser -v testdb
```

**Comment by lkshminarayanan** [comment]
There is an issue upstream that fixes this issue by patching `pg_dump` - https://www.postgresql.org/message-id/flat/00d46a48-3324-d9a0-49bf-e7f0f11d1038%40timescale.com

**Comment by a-nldisr** [comment]
Issue is also present in MST

**Comment by lkshminarayanan** [comment]
Hi @a-nldisr! Can you give me more information on the error you are facing? There are workarounds in place in MST and Timescale cloud to prevent this from happening, so I'm not sure why you might be facing this.

**Comment by estokes-vs** [comment]
@lkshminarayanan I ran into this issue today in MST trying to take a backup on our database freshly upgraded to Timescale 2.11.  What workarounds are you referring to for MST?

I followed a couple breadcrumbs here: https://github.com/timescale/docs/issues/2226

However when I tried running the GRANT permission I received a warning that no permissions had been granted and the `pg_dump` still fails.  We are required for security compliance to have multiple locations for backups, so running these with `pg_dump` is essential for our use case. 

```
db=> GRANT SELECT ON TABLE _timescaledb_internal.job_errors TO db_backup;
WARNING:  no privileges were granted for "job_errors"
GRANT
```

Thanks for all the help so far in identifying this and working towards a resolution
